app/models/chargeback.rb

```diff
@@ -117,28 +158,31 @@ def get_rates(perf)
     @rates[key] = ChargebackRate.get_assigned_for_target(perf.resource, :tag_list => tag_list, :parents => parents)
   end
 
-  def self.calculate_costs(perf, h, rates)
-    # This expects perf interval to be hourly. That will be the most granular interval available for chargeback.
-    unless perf.capture_interval_name == "hourly"
-      raise _("expected 'hourly' performance interval but got '%{interval}") % {:interval => perf.capture_interval_name}
-    end
+  def self.calculate_costs(metric_rollup_records, rates, hours_in_interval)
+    calculated_costs = {}
 
-    calc_fixed_compute = calc_fixed_compute?(perf)
-    h['fixed_compute_metric'] = (h['fixed_compute_metric'] || 0) + 1 if calc_fixed_compute
+    chargeback_fields_present                = metric_rollup_records.count(&:chargeback_fields_present?)
+    calculated_costs['fixed_compute_metric'] = chargeback_fields_present if chargeback_fields_present
 
     rates.each do |rate|
       rate.chargeback_rate_details.each do |r|
-        rec    = r.metric && perf.respond_to?(r.metric) ? perf : perf.resource
-        metric = r.metric.nil? ? 0 : rec.send(r.metric) || 0
-        cost   = r.group == 'fixed' && !calc_fixed_compute ? 0 : r.cost(metric)
+        if !chargeback_fields_present && r.fixed?
+          cost = 0
+        else
+          metric_value = r.metric_value_by(metric_rollup_records)
+          cost = r.cost(metric_value) * hours_in_interval
```

Comment:
@gtanzillo I move out multiplying of cost method because we don't need it calculate (because we are not using accuntable period)   separately fixed and variable part.